src/main/java/com/arpnetworking/tsdcore/sinks/HttpSinkActor.java

```diff
@@ -257,12 +273,24 @@ private void dispatchPending() {
     }
 
     private void fireNextRequest() {
-        final Request request = _pendingRequests.poll();
+        final RequestEntry requestEntry = _pendingRequests.poll();
+        final Metrics metrics = _metricsFactory.create();
+        metrics.setTimer(_inQueueLatencyName, requestEntry.getEnterTime() - System.currentTimeMillis(), TimeUnit.MILLISECONDS);
+
+        final Request request = requestEntry.getRequest();
         _inflightRequestsCount++;
 
         final CompletableFuture<Response> promise = new CompletableFuture<>();
+        metrics.startTimer(_requestLatencyName);
         _client.executeRequest(request, new ResponseAsyncCompletionHandler(promise));
         final CompletionStage<Object> responsePromise = promise
+                .whenComplete((result, err) -> {
+                        metrics.stopTimer(_requestLatencyName);
+                        if (err != null) {
+                            metrics.incrementCounter(_requestSuccessName, 0);
```

Comment:
I am concerned that we've split success measurement across two methods. That makes it rather brittle.